tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -1369,28 +1366,79 @@ def test_search_runs_returns_expected_results_with_large_experiment(self):
+                    'key': 'mkey_0',
+                    'value': current_run,
+                    'timestamp': 100 * 2,
+                    'step': 100 * 3,
+                    'is_nan': 0,
+                    'run_uuid': run_id,
+                }
+            )
+            current_run += 1
         metrics = pd.DataFrame(metrics_list)
         metrics.to_sql('metrics', self.store.engine, if_exists='append', index=False)
         params = pd.DataFrame(params_list)
         params.to_sql('params', self.store.engine, if_exists='append', index=False)
         tags = pd.DataFrame(tags_list)
         tags.to_sql('tags', self.store.engine, if_exists='append', index=False)
+        pd.DataFrame(latest_metrics_list).to_sql(
+            'latest_metrics', self.store.engine, if_exists='append', index=False)
+        return experiment_id, run_ids
+
+    def test_search_runs_returns_expected_results_with_large_experiment(self):
+        """
+        This case tests the SQLAlchemyStore implementation of the SearchRuns API to ensure
+        that search queries over an experiment containing many runs, each with a large number
+        of metrics, parameters, and tags, are performant and return the expected results.
+        """
+        experiment_id, run_ids = self._generate_large_data()
 
         run_results = self.store.search_runs([experiment_id], None, ViewType.ALL, max_results=100)
-        assert len(run_results) > 0
+        assert len(run_results) == 100
         assert set([run.info.run_id for run in run_results]).issubset(set(run_ids))
 
+    def test_search_runs_correctly_filters_large_data(self):
+        experiment_id, _ = self._generate_large_data(1000)
+
+        run_results = self.store.search_runs([experiment_id],
+                                             "metrics.mkey_0 < 26 and metrics.mkey_0 > 5 ",
+                                             ViewType.ALL, max_results=1000)
+        assert len(run_results) == 20
+
+        run_results = self.store.search_runs([experiment_id],
+                                             "metrics.mkey_0 < 26 and metrics.mkey_0 > 5 "
+                                             "and tags.tkey_0 = 'tval_0' ",
+                                             ViewType.ALL, max_results=1000)
+        assert len(run_results) == 2  # 20 runs between 9 and 26, 2 of which have a 0 tkey_0 value
+
+        run_results = self.store.search_runs([experiment_id],
+                                             "metrics.mkey_0 < 26 and metrics.mkey_0 > 5 "
+                                             "and tags.tkey_0 = 'tval_0' "
+                                             "and params.pkey_0 = 'pval_0'",
+                                             ViewType.ALL, max_results=1000)
+        assert len(run_results) == 1  # 2 runs on previous request, 1 of which has a 0 pkey_0 value
+
+    def test_get_attribute_name(self):
+        assert(models.SqlRun.get_attribute_name("artifact_uri") == "artifact_uri")
```

Comment:
@jcuquemelle I think the following test definition is more thorough. Please let me know what you think!:

```
def test_get_attribute_name():
    for attribute_name in SearchUtils.VALID_SEARCH_ATTRIBUTE_KEYS:
        assert hasattr(models.SqlRun, models.SqlRun.get_attribute_name(attribute_name))
```

Reply:
I thought of doing that, but I prefer explicitely expose the mapping (and break if the mapping has a different size than the length of VALID_SEARCH_ATTRIBUTE_KEYS) because an attribute could exist but not be the one we want to map to, e.g. in case of deprecation. 
We already have a case (that is not searchable, but illustrates my point): run_data.metrics maps to SQLrun.latest_metrics whereas SQLRun has a .metrics attibute.